### ISSUE TITLE
Compile server should not be directly accessible.

### ISSUE THREAD

**Issue by richardpawson** [MetalUp/Express#34]
(This should probably be left until we have switched to a Windows job server.)

Either we find some way to ensure that the compile server can only be accessed via the MetalUp.Express client, OR channel all requests from the client through the MetalUp server.

Comment from PDD:  At the moment it looks like the jobe server is accepting requests straight from the client. This means that all validation is done client side and could potentially be overridden / exploited. If possible in the future, I'd recommend implementing an API so that the MetalUp.Express client submits expressions / functions to the metalup server which then validates the code before constructing the JSON data to send to the jobe server. This means that you can set up a firewall so that the jobe server only communicates with your MetalUp host rather than the client directly (also blocking any other outbound traffic)

Bottom line: it should not be possible for someone to access the compile server directly.

**Comment by richardpawson** [comment]
Superseded/duplicated by #86
